shoplist/src/main/java/com/netguru/codereview/ui/MainViewModel.kt

```diff
@@ -8,33 +8,51 @@ import com.netguru.codereview.network.ShopListApiMock
 import com.netguru.codereview.network.ShopListRepository
 import com.netguru.codereview.network.model.ShopListItemResponse
 import com.netguru.codereview.network.model.ShopListResponse
-import kotlinx.coroutines.GlobalScope
+import com.netguru.codereview.ui.model.ResultState
+import com.netguru.codereview.ui.model.ShopList
 import kotlinx.coroutines.flow.collect
 import kotlinx.coroutines.launch
 
 class MainViewModel : ViewModel() {
 
     private val shopListRepository = ShopListRepository(ShopListApiMock())
 
-    val shopLists = MutableLiveData<List<Pair<ShopListResponse, List<ShopListItemResponse>>>>()
-    private val eventLiveData = MutableLiveData<String>()
+    private val _shopListsLiveData =
+        MutableLiveData<ResultState<List<ShopList?>?>>()
+    internal val shopListsLiveData: LiveData<ResultState<List<ShopList?>?>>
+        get() = _shopListsLiveData
```

Comment:
not expose mutable type, https://developer.android.com/kotlin/coroutines/coroutines-best-practices#mutable-types
All changes of mutable types happened in one class to make it easy to debug when something happened